src/AreaView/AreaCard/Details/AreaCardDetailsEdit.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from "react";
+import { MutableField } from "../../../MutableField";
+import * as Actions from "../Provider/AreaCardActions";
+import { Button } from "azure-devops-ui/Button";
+import { useAreaCardValue } from "../Provider/AreaCardProvider";
+import { useStateValue, IOKRContext } from "../../../StateMangement/StateProvider";
+
+export const AreaCardDetailsEdit: React.FunctionComponent = props => {
+	const stateContext = useStateValue();
+	const [{ area }, areaDispatcher] = useAreaCardValue();
+
+	return (
+		<div className="card-header">
+			<h3>
+				<MutableField value={area.Name} onChange={(val) => { area.Name = val; }} />
```

Comment:
What state value are you updating here? How does the cancel work?

Reply:
If this is directly updating the area value from useAreaCardValue, that's the "global" state, (only for this dispatcher, but still "global"). When the reducer is fired to set the area = action.area, isn't that already set?

Reply:
Forgot to do Name. Good catch.

Reply:
As for cancel, I was getting it for free from MutableField (it only fires onChange when it loses focus, and it discards on escape).